utils/atomics.go

```diff
@@ -0,0 +1,46 @@
+package utils
```

Comment:
how about `misc` then? :) he he...

I agree needs refactored at some point! For now I've just been dumping things here...